core/src/main/java/org/polypheny/db/catalog/impl/logical/GraphCatalog.java

```diff
@@ -0,0 +1,116 @@
+ * limitations under the License.
+ */
+
+package org.polypheny.db.catalog.impl.logical;
+
+import io.activej.serializer.BinarySerializer;
+import io.activej.serializer.annotations.Deserialize;
+import io.activej.serializer.annotations.Serialize;
+import java.beans.PropertyChangeSupport;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import lombok.Getter;
+import lombok.Value;
+import lombok.experimental.SuperBuilder;
+import org.polypheny.db.catalog.Catalog;
+import org.polypheny.db.catalog.IdBuilder;
+import org.polypheny.db.catalog.catalogs.LogicalCatalog;
+import org.polypheny.db.catalog.catalogs.LogicalGraphCatalog;
+import org.polypheny.db.catalog.entity.logical.LogicalGraph;
+import org.polypheny.db.catalog.entity.logical.LogicalNamespace;
+import org.polypheny.db.type.PolySerializable;
+
+@Value
+@SuperBuilder(toBuilder = true)
+public class GraphCatalog implements PolySerializable, LogicalGraphCatalog {
+
+    @Getter
+    public BinarySerializer<GraphCatalog> serializer = PolySerializable.buildSerializer( GraphCatalog.class );
+    @Getter
+    @Serialize
+    public LogicalNamespace logicalNamespace;
+    public IdBuilder idBuilder = IdBuilder.getInstance();
+
+    @Getter
+    @Serialize
+    public ConcurrentHashMap<Long, LogicalGraph> graphs;
+
+
+    public GraphCatalog( LogicalNamespace logicalNamespace ) {
+        this( logicalNamespace, new ConcurrentHashMap<>() );
+    }
+
+
+    public GraphCatalog(
+            @Deserialize("logicalNamespace") LogicalNamespace logicalNamespace,
+            @Deserialize("graphs") Map<Long, LogicalGraph> graphs ) {
+        this.logicalNamespace = logicalNamespace;
+        this.graphs = new ConcurrentHashMap<>( graphs );
+
+        // already add only graph
+        addGraph( logicalNamespace.id, logicalNamespace.name, true );
+        listeners.addPropertyChangeListener( Catalog.getInstance().getChangeListener() );
+    }
+
+
+    PropertyChangeSupport listeners = new PropertyChangeSupport( this );
+
+
+    public void change() {
+        listeners.firePropertyChange( "change", null, null );
+    }
+
+
+    @Override
+    public PolySerializable copy() {
+        return PolySerializable.deserialize( serialize(), GraphCatalog.class );
+    }
+
+
+    @Override
+    public LogicalCatalog withLogicalNamespace( LogicalNamespace namespace ) {
+        return toBuilder().logicalNamespace( namespace ).build();
+    }
+
+
+    @Override
+    public void addGraphAlias( long graphId, String alias, boolean ifNotExists ) {
+        change();
+    }
+
+
+    @Override
+    public void removeGraphAlias( long graphId, String alias, boolean ifExists ) {
+        change();
+    }
+
+
+    @Override
+    public LogicalGraph addGraph( long id, String name, boolean modifiable ) {
+        LogicalGraph graph = new LogicalGraph( id, name, modifiable, logicalNamespace.caseSensitive );
+        graphs.put( id, graph );
+        change();
```

Comment:
This fires change without specifying what changed

Reply:
https://github.com/polypheny/Polypheny-DB/pull/464#discussion_r1518560462